app/forms/candidate_interface/english_gcse_grade_form.rb

```diff
@@ -0,0 +1,276 @@
+                  :english_double_award,
+                  :grade_english_double,
+                  :english_language,
+                  :grade_english_language,
+                  :english_literature,
+                  :grade_english_literature,
+                  :english_studies_single_award,
+                  :grade_english_studies_single,
+                  :english_studies_double_award,
+                  :grade_english_studies_double,
+                  :other_english_gcse,
+                  :other_english_gcse_name,
+                  :grade_other_english_gcse
+
+    validates :grade, presence: true, on: :grade
+    validates :other_grade, presence: true, if: :grade_is_other?
+    validate :validate_grade_format, on: :grade, unless: :is_multiple_gcse? || :new_record?
+    validate :validate_grades, unless: :new_record?, on: :grades, if: :is_multiple_gcse?
+    validate :gcse_selected, on: :grades, if: :is_multiple_gcse?
+
+    class << self
+      def build_from_qualification(qualification)
+        if qualification.qualification_type == 'non_uk'
+          new(grade: qualification.set_grade,
+              other_grade: qualification.set_other_grade,
+              qualification: qualification)
+        else
+          new(build_params_from(qualification))
+        end
+      end
+
+    private
+
+      def build_params_from(qualification)
+        params = {
+          grade: qualification.grade,
+          qualification: qualification,
+        }
+
+        if qualification.grades
+          grades = JSON.parse(qualification.grades)
+          english_gcses = []
+
+          if grades.keys.include? 'english_single_award'
+            english_gcses << 'english_single_award'
+            params[:grade_english_single] = grades['english_single_award']
+            params[:english_single_award] = true
+          end
+
+          if grades.keys.include? 'english_double_award'
+            english_gcses << 'english_double_award'
+            params[:grade_english_double] = grades['english_double_award']
+            params[:english_double_award] = true
+          end
+
+          if grades.keys.include? 'english_language'
+            english_gcses << 'english_language'
+            params[:grade_english_language] = grades['english_language']
+            params[:english_language] = true
+          end
+
+          if grades.keys.include? 'english_literature'
+            english_gcses << 'english_literature'
+            params[:grade_english_literature] = grades['english_literature']
+            params[:english_literature] = true
+          end
+
+          if grades.keys.include? 'english_studies_single_award'
+            english_gcses << 'english_studies_single_award'
+            params[:grade_english_studies_single] = grades['english_studies_single_award']
+            params[:english_studies_single_award] = true
+          end
+
+          if grades.keys.include? 'english_studies_double_award'
```

Comment:
The double award validation for English studies did not kick in for me.
<img width="973" alt="english_studies_validation" src="https://user-images.githubusercontent.com/5256922/99401074-a9694a80-28df-11eb-8d09-62db08df75e4.png">

Reply:
Thanks, I've found the typo and resolved that now